File: src/runtime-info.ts
/**
 * Read-only facts about the current terminal process.
 */
import { getProviderEnvMap, type Store } from './config/store.js';
import { providerDisplayName, T } from './i18n/index.js';

/** Key-safe description of the API currently visible to this terminal process. */
export function currentTerminalLine(store: Store): string {
  return T('terminal.current', currentTerminalTarget(store));
}

function currentTerminalTarget(store: Store): string {
  const base = (process.env.ANTHROPIC_BASE_URL ?? '').trim();
  if (!base) return T('terminal.official');

  const host = hostOf(base);
  const match = store.providers.find((p) => sameBase(base, getProviderEnvMap(p).ANTHROPIC_BASE_URL ?? ''));
  if (match) return T('terminal.matched', host, providerDisplayName(match));
  return T('terminal.unmatched', host);
}

function sameBase(a: string, b: string): boolean {
  return a.trim().replace(/\/+$/, '') === b.trim().replace(/\/+$/, '');
}

function hostOf(raw: string): string {
  try {
    const u = new URL(raw);
    return u.host || raw;
  } catch {
    return raw;
  }
}

File: internal/runtimeinfo/runtimeinfo.go
// Package runtimeinfo formats read-only facts about the current terminal.
package runtimeinfo

import (
	"net/url"
	"os"
	"strings"

	"github.com/becomeless/cc-x/internal/config"
	"github.com/becomeless/cc-x/internal/i18n"
)

// CurrentTerminalLine returns a localized, key-safe description of the API
// currently visible to this terminal process.
func CurrentTerminalLine(store *config.Store) string {
	return i18n.T("terminal.current", currentTerminalTarget(store))
}

func currentTerminalTarget(store *config.Store) string {
	base := strings.TrimSpace(os.Getenv("ANTHROPIC_BASE_URL"))
	if base == "" {
		return i18n.T("terminal.official")
	}

	host := hostOf(base)
	for _, p := range store.Providers {
		pbase := strings.TrimSpace(config.GetProviderEnvMap(p)["ANTHROPIC_BASE_URL"])
		if sameBase(base, pbase) {
			return i18n.T("terminal.matched", host, i18n.ProviderDisplayName(p))
		}
	}
	return i18n.T("terminal.unmatched", host)
}

func sameBase(a, b string) bool {
	return strings.TrimRight(strings.TrimSpace(a), "/") == strings.TrimRight(strings.TrimSpace(b), "/")
}

func hostOf(raw string) string {
	u, err := url.Parse(raw)
	if err == nil && u.Host != "" {
		return u.Host
	}
	return raw
}
